### ISSUE TITLE
HSTS support?

### ISSUE THREAD

**Issue by equal-l2** [SergioBenitez/Rocket#664]
I'm working on a HTTPS site with Rocket and I wonder if Rocket can redirect visitors from HTTP to HTTPS.
For example, when a visitor come to `http://example.com/`, then I want Rocket to redirect one to `https://example.com/`.
I've googled and found that [HSTS](https://en.wikipedia.org/wiki/HTTP_Strict_Transport_Security) can do that. 
Does Rocket supports HSTS, or more simply HTTP-to-HTTPS redirection?

**Comment by jebrosen** [comment]
HSTS itself only directs a browser to automatically upgrade all *future* connections to HTTPS, and it can only be set over HTTPS. You still would need to do some redirection from HTTP to HTTPS in the first place.

If the site is served via nginx, `return 301 https://$host$request_uri;` can be used in the http server block. Similar approaches should work for other servers.

If the goal is to use only Rocket, this can be accomplished with two rocket instances: one on HTTPS that serves the site, and another on HTTP that only ever responds with a redirect to HTTPS.

**Comment by pfpacket** [comment]
@jebrosen 

> If the goal is to use only Rocket, this can be accomplished with two rocket instances: one on HTTPS that serves the site, and another on HTTP that only ever responds with a redirect to HTTPS.

How can we launch a HTTP server with the `tls` feature enabled? (I mean launching HTTP and HTTPS servers in the same program)
The only way I came up with to accomplish it was to make a new program that just launches a HTTP server to redirect to HTTPS.

**Comment by jebrosen** [comment]
You can use a different configuration for the HTTP server and use [`Rocket::custom`](https://api.rocket.rs/rocket/struct.Rocket.html#method.custom) instead of `Rocket::ignite`. In particular, you could use a "blank" config which will have no tls settings.

This situation seems common enough (and you're not the first to ask), so it might be worth an example or an addendum to the guide.

**Comment by pfpacket** [comment]
@jebrosen Yes, you are absolutely right! I thought uncommenting `certs` and `key` in `global.tls` was enough to disable TLS and it wasn't (I should have uncommented the whole `global.tls` table). That was why I thought I couldn't disable it with the `tls` feature enabled.

Thanks.
